storage_controller/src/compute_hook.rs

```diff
@@ -165,6 +187,7 @@ struct ComputeHookNotifyRequestShard {
 #[derive(Serialize, Deserialize, Debug, Eq, PartialEq)]
 struct ComputeHookNotifyRequest {
     tenant_id: TenantId,
+    preferred_az: Option<String>,
```

Comment:
@mattpodraza FYI - I'm adding this here so that in future we have the option of consuming it in the control plane (extra JSON fields won't break any existing decoding, right?).  This would enable getting the preferred AZ more reliably than inferring it from shard 0's location.